engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/PatternValidator.java

```diff
@@ -8,35 +8,40 @@
 import java.util.regex.Matcher;
 import java.util.regex.PatternSyntaxException;
 
-import jakarta.validation.ConstraintValidator;
 import jakarta.validation.ConstraintValidatorContext;
 import jakarta.validation.constraints.Pattern;
+import jakarta.validation.metadata.ConstraintDescriptor;
 
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidator;
 import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorInitializationContext;
+import org.hibernate.validator.internal.engine.constraintvalidation.PatternConstraintInitializer;
 import org.hibernate.validator.internal.engine.messageinterpolation.util.InterpolationHelper;
 import org.hibernate.validator.internal.util.logging.Log;
 import org.hibernate.validator.internal.util.logging.LoggerFactory;
 
 /**
  * @author Hardy Ferentschik
  */
-public class PatternValidator implements ConstraintValidator<Pattern, CharSequence> {
+public class PatternValidator implements HibernateConstraintValidator<Pattern, CharSequence> {
 
 	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
 
 	private java.util.regex.Pattern pattern;
 	private String escapedRegexp;
 
 	@Override
-	public void initialize(Pattern parameters) {
+	public void initialize(ConstraintDescriptor<Pattern> constraintDescriptor, HibernateConstraintValidatorInitializationContext initializationContext) {
+		Pattern parameters = constraintDescriptor.getAnnotation();
 		Pattern.Flag[] flags = parameters.flags();
 		int intFlag = 0;
 		for ( Pattern.Flag flag : flags ) {
 			intFlag = intFlag | flag.getValue();
 		}
 
 		try {
-			pattern = java.util.regex.Pattern.compile( parameters.regexp(), intFlag );
+			pattern = initializationContext.getConstraintValidatorInitializationSharedService( PatternConstraintInitializer.class )
```

Comment:
I thought ... that shared service might be a good fit for the name 🙂 🫣

Reply:
I would say that a service is generally stateless, but YMMV :)

If `cache` really rubs you the wrong way, maybe `sharedData`?

By the way, why `getConstraintValidatorInitialization*`? This is a method in `initializationContext`, so `get*` would be enough, no?